dataclients/kubernetes/kube.go

```diff
@@ -1356,3 +1386,54 @@ func (c *Client) Close() {
 		close(c.quit)
 	}
 }
+
+func (c *Client) fetchDefaultFilterConfigs() map[resourceId]string {
+	if c.defaultFiltersDir == "" {
+		log.Debug("default filters are disabled")
+		return make(map[resourceId]string)
+	}
+
+	filters, err := c.getDefaultFilterConfigurations()
+
+	if err != nil {
+		log.WithError(err).Error("could not fetch default filter configurations")
+		return make(map[resourceId]string)
+	}
+
+	log.WithField("#configs", len(filters)).Debug("default filter configurations loaded")
+
+	return filters
+}
+
+func (c *Client) getDefaultFilterConfigurations() (map[resourceId]string, error) {
+	files, err := ioutil.ReadDir(c.defaultFiltersDir)
+	if err != nil {
+		return nil, err
+	}
+
+	filters := make(map[resourceId]string)
+	for _, f := range files {
+		r := strings.Split(f.Name(), ".") // format: {service}.{namespace}
+		if len(r) != 2 || notRegularFile(f) || f.Size() > maxFileSize {
+			log.WithError(err).WithField("file", f.Name()).Debug("incompatible file")
+			continue
+		}
+
+		file := filepath.Join(c.defaultFiltersDir, f.Name())
+		config, err := ioutil.ReadFile(file)
+		if err != nil {
+			log.WithError(err).WithField("file", file).Debug("could not read file")
+			continue
+		}
+
+		filters[resourceId{name: r[0], namespace: r[1]}] = string(config)
+	}
+
+	return filters, nil
+}
+
+func notRegularFile(f os.FileInfo) bool {
+	mode := f.Mode()
+	return f.IsDir() || mode == os.ModeIrregular || mode == os.ModeSymlink || mode == os.ModeDevice ||
```

Comment:
q: what is the reason we're not allowing symlinks here?

Reply:
Actually, we have to allow symlinks. Good catch, thank you!